addons/mail/static/src/core/common/thread.xml

```diff
@@ -22,7 +23,11 @@
                         <DateSection date="msg.dateDay" className="'pt-4'"/>
                         <t t-set="currentDay" t-value="msg.dateDay"/>
                     </t>
-                    <Transition t-if="props.thread.model === 'discuss.channel'"  visible="props.thread.seen_message_id === (prevMsg?.id ?? false)" name="'o-fade'" t-slot-scope="transition">
+                    <t t-set="showSeparator" t-value="(!msg.isSelfAuthored or props.thread.seen_message_id === (prevMsg?.id ?? false)) and (msg.id &gt; props.thread.seen_message_id and !separatorDisplayed)" />
+                    <t t-if="showSeparator">
+                        <t t-set="separatorDisplayed" t-value="true"/>
+                    </t>
+                    <Transition t-if="props.thread.model === 'discuss.channel'" visible="showSeparator" name="'o-fade'" t-slot-scope="transition">
```

Comment:
1. Thread order is not necessarily `asc`. Current implementation only kinda work in `asc`.
2. Condition doesn't look correct:
- if `props.thread.seen_message_id` is `false` (which is a genuine value to say no `seen_message_id`), and 1st message is not self authored, then:
- `t-value` is `(!false or ...) and (1 > false and !false)` so `true and (true and true)` thus `true` and it displays "new message" when it shouldn't.
3. All this processing when this is ignored if the current thread is not a `discuss.channel`. Accessing `this.props.thread.seen_message_id` or `message.id` is not cheap due to `owl.reactive` accessors being slow.

Reply:
> if props.thread.seen_message_id is false (which is a genuine value to say no seen_message_id), and 1st message is not self authored, then:
> t-value is (!false or ...) and (1 > false and !false) so true and (true and true) thus true and it displays "new message" when it shouldn't.

shouldn't we show message separator if there is at least one unseen message which is not self authored ?

Reply:
> shouldn't we show message separator if there is at least one unseen message which is not self authored ?

`(!msg.isSelfAuthored or props.thread.seen_message_id === (prevMsg?.id ?? false))` is truthy as long as `!msg.isSelfAuthored` (= message is not self authored). The whole condition is truthy while there's no unseen message (that's the `this.props.thread.seen_message_id`, which I hypothetically set to `false` in this scenario and the condition is evaluated to `true`)

Reply:
But doesn't `seen_message_id = false` means that the member hasn't seen any messages? As this field is fetched from DB it should not have any guessed value. It should be accurate.

Reply:
`seen_message_id` should have a good default, and a good default is for the feature to not work, i.e. by not showing new message separator. If we want to say `seen_message_id` is the earliest, it should have an explicit value, whether `0` or `-1`. Note that `seen_message_id` is relational field in python and `None` becomes `false` through JSON data, so web client sees `false`.

Good defaults are important to not assume things are necessarily known (e.g. it hasn't necessarily been fetched from server) nor they necessarily work in all contexts (e.g. it doesn't apply in chatter). We are currently very greedy in fetched data of channel members and lots (too many) chatter/channel branching to prevent misuse of `seen_message_id` in chatter. Good defaults result in less bugs and less code to maintain :)

Reply:
@rapr-odoo Sorry for the confusion, in `17.0` `seen_message_id = false` actually means hasn't seen any message and the feature is enabled. So you were right and I was wrong. I was confused with the modeling in `saas-17.3` and after, were `seen_message_id` was decoupled with seen feature and has been replaced with `new_message_separator`, whose value follows the distinction of `0`/`false` that I said.

So in the end, `seen_message_id = false` in 17.0 necessarily means show seen message separator when there are some non-self authored messages. It's therefore ok to show it on the 1st non-self authored message.

Again sorry for my mistake in assuming `false` means "not applicable" 🙈